client/transaction-pool/src/metrics.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Transaction pool prometheus metrics.
+
+use prometheus_endpoint::{register, Counter, Gauge, PrometheusError, Registry, U64};
+
+/// Transaction pool prometheus metrics.
+pub struct Metrics {
+	pub validation_pending: Gauge<U64>,
+	pub total_validated: Counter<U64>,
+}
+
+
+impl Metrics {
+	pub fn register(registry: &Registry) -> Result<Self, PrometheusError> {
+		Ok(Self {
+			validation_pending: register(
```

Comment:
Instead of using a single `Gauge` here I would suggest using two monotonically increasing `Counter`s.

Using a single `Gauge` might miss spikes. E.g. imagine the following scenario:

1. Prometheus collects the metrics.

2. 99999999 transactions are scheduled for validation.

3. 99999999 are done validating.

4. Prometheus collects metrics.

Using two `Counter`s increasing one when a new transaction is scheduled for validation and increasing another when a transaction is done validating.

What do you think @NikVolf?

Reply:
Yes, I got it, redid with two counters